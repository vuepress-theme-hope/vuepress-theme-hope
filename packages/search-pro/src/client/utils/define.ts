import type {
  SearchProClientCustomFiledConfig,
  SearchProLocaleConfig,
} from "../../shared/index.js";

declare const SEARCH_PRO_CUSTOM_FIELDS: SearchProClientCustomFiledConfig;
declare const SEARCH_PRO_LOCALES: SearchProLocaleConfig;

export const searchProLocales = SEARCH_PRO_LOCALES;
export const searchProClientCustomFiledConfig = SEARCH_PRO_CUSTOM_FIELDS;
