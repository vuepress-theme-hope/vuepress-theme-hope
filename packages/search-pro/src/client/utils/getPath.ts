import { store } from "@temp/search-pro/store.js";

import type { MatchedItem } from "../typings/index.js";

export const getPath = (item: MatchedItem): string =>
  store[item.id] + ("anchor" in item ? `#${item.anchor}` : "");
