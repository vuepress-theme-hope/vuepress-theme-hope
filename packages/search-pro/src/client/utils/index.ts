export * from "./closeIcon.js";
export * from "./getPath.js";
export * from "./isFocusingTextControl.js";
export * from "./isKeyMatched.js";
export * from "./keyIcons.js";
