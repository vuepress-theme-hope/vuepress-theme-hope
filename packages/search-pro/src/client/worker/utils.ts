import type { SearchOptions } from "../typings/index.js";

const CHINESE_CHARACTERS = /[\u4e00-\u9fa5]/g;

export const getSearchOptions = (
  searchOptions: SearchOptions = {}
): SearchOptions => ({
  fuzzy: 0.2,
  prefix: true,
  processTerm: (term): string[] => {
    const chineseCharacters = term.match(CHINESE_CHARACTERS) || [];
    const englishTerm = term.replace(CHINESE_CHARACTERS, "").toLowerCase();

    return englishTerm
      ? [englishTerm, ...chineseCharacters]
      : [...chineseCharacters];
  },
  ...searchOptions,
});
