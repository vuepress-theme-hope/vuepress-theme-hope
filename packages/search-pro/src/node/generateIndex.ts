import { type App, type Page } from "@vuepress/core";
import { isArray } from "@vuepress/shared";
import { type AnyNode, load } from "cheerio";
import { fromEntries, keys } from "vuepress-shared/node";

import {
  type SearchProCustomFieldOptions,
  type SearchProOptions,
} from "./options.js";
import {
  type PageHeaderContent,
  type PageIndex,
  type SearchIndex,
} from "../shared/index.js";

/**
 * These tags are valid HTML tags which can contain content.
 */

/**
 * @description h1 is removed because it's the title of the page.
 */
const HEADING_TAGS = "h2,h3,h4,h5,h6".split(",");

/**
 * @description Not all the block tags are included, because some of them shall not be indexed
 */
const CONTENT_BLOCK_TAGS =
  "header,nav,section,div,dd,dl,dt,figcaption,figure,picture,hr,li,main,ol,p,ul,caption,table,thead,tbody,th,tr,td,datalist,fieldset,form,legend,optgroup,option,select,details,dialog,menu,menuitem,summary,blockquote,tfoot".split(
    ","
  );

/**
 * @description Not all the inline tags are included, because some of them shall not be indexed, e.g.: pre
 *
 * routerlink is added to the list, because it is a special link tag
 */
const CONTENT_INLINE_TAGS =
  "routerlink,a,b,abbr,bdi,bdo,cite,code,dfn,em,i,kbd,mark,q,rp,rt,ruby,s,samp,small,span,strong,sub,sup,time,u,var,wbr,del,ins,button,label,legend,meter,optgroup,option,output,progress,select".split(
    ","
  );

const $ = load("");

export const generatePageIndex = (
  page: Page<{ excerpt?: string }>,
  customFieldsGetter: SearchProCustomFieldOptions[] = [],
  indexContent = false
): PageIndex | null => {
  const hasExcerpt = "excerpt" in page.data && page.data["excerpt"].length;

  const result: PageIndex = {
    title: page.title,
    contents: [],
  };

  // here are some variables holding the current state of the parser
  let shouldIndexContent = hasExcerpt || indexContent;
  let currentContent = "";
  let currentHeaderContent: PageHeaderContent = {
    header: "",
    slug: "",
    contents: [],
  };
  let isContentBeforeFirstHeader = true;

  const render = (node: AnyNode): void => {
    if (node.type === "tag") {
      if (HEADING_TAGS.includes(node.name)) {
        if (currentContent && shouldIndexContent) {
          // add last content
          currentHeaderContent?.contents.push(
            currentContent.replace(/\s+/gu, " ")
          );
          currentContent = "";
        }

        // content before first header does not belong to any header
        if (isContentBeforeFirstHeader) {
          // the content before the first header shall have actual contents
          if (currentHeaderContent.contents.length)
            result.contents.push(currentHeaderContent);

          isContentBeforeFirstHeader = false;
        } else {
          result.contents.push(currentHeaderContent);
        }

        // update header
        currentHeaderContent = {
          header: node.children
            .map((node) => (node.type === "text" ? node.data : ""))
            .join("")
            .trim(),
          slug: node.attribs["id"],
          contents: [],
        };
      } else if (CONTENT_BLOCK_TAGS.includes(node.name)) {
        if (currentContent && shouldIndexContent) {
          // add last content
          currentHeaderContent?.contents.push(
            currentContent.replace(/\s+/gu, " ")
          );
          currentContent = "";
        }
        node.childNodes.forEach(render);
      } else if (CONTENT_INLINE_TAGS.includes(node.name)) {
        node.childNodes.forEach(render);
      }
    } else if (node.type === "text") {
      currentContent += node.data.trim() ? node.data : "";
    } else if (
      // we are expecting to stop at excerpt marker
      hasExcerpt &&
      !indexContent &&
      // we got excerpt marker
      node.type === "comment" &&
      node.data.trim() === "more"
    ) {
      shouldIndexContent = false;
    }
  };

  const nodes = $.parseHTML(page.contentRendered);

  // get custom fields
  const customFields = fromEntries(
    customFieldsGetter
      .map(({ getter }, index) => {
        const result = getter(page);

        return isArray(result)
          ? [index.toString(), result]
          : result
          ? [index.toString(), [result]]
          : null;
      })
      .filter((item): item is [string, string[]] => item !== null)
  );

  // no content in page and no customFields
  if (!nodes?.length && !keys(customFields).length) return null;

  // walk through nodes and extract indexes
  nodes?.forEach((node) => {
    render(node);
  });

  // push contents in last block tags
  if (shouldIndexContent && currentContent)
    currentHeaderContent?.contents.push(currentContent);

  // push last content
  if (currentHeaderContent.contents.length)
    result.contents.push(currentHeaderContent);

  return {
    ...result,
    ...(keys(customFields).length ? { customFields } : {}),
  };
};

export const getSearchIndex = (
  app: App,
  options: SearchProOptions
): SearchIndex => {
  const pagesSearchIndex = app.pages
    .map((page) => {
      const pageIndex = generatePageIndex(
        page,
        options.customFields,
        options.indexContent
      );

      return pageIndex
        ? { path: page.path, index: pageIndex, localePath: page.pathLocale }
        : null;
    })
    .filter(
      (item): item is { path: string; index: PageIndex; localePath: string } =>
        item !== null
    );

  return fromEntries(
    keys(
      // locales should at least have root locales
      // eslint-disable-next-line @typescript-eslint/naming-convention
      { "/": {}, ...app.options.locales }
    ).map((localePath) => [
      localePath,
      fromEntries(
        pagesSearchIndex
          .filter((item) => item.localePath === localePath)
          .map((item) => [item.path, item.index])
      ),
    ])
  );
};
