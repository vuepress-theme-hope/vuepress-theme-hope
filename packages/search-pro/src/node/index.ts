export * from "./options.js";
export * from "./plugin.js";
export * from "../shared/index.js";
