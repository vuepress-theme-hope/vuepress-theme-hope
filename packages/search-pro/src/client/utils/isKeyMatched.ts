import { searchProHotKeys } from "../define.js";

export const isKeyMatched = (event: KeyboardEvent): boolean =>
  searchProHotKeys.some((item) => {
    const { key, ctrl = false, shift = false, alt = false } = item;

    return (
      key === event.key &&
      ctrl === event.ctrlKey &&
      shift === event.shiftKey &&
      alt === event.altKey
    );
  });
