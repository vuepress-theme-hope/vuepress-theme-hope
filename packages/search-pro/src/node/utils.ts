import { ensureEndingSlash } from "@vuepress/shared";
import { getDirname, path } from "@vuepress/utils";
import { Logger } from "vuepress-shared/node";

const __dirname = getDirname(import.meta.url);

export const CLIENT_FOLDER = ensureEndingSlash(
  path.resolve(__dirname, "../client/")
);

export const WORKER_FOLDER = ensureEndingSlash(
  path.resolve(__dirname, "../worker/")
);

export const PLUGIN_NAME = "vuepress-plugin-search-pro";

export const logger = new Logger(PLUGIN_NAME);
