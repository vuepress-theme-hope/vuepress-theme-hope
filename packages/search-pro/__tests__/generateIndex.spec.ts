import { type Page, createBaseApp } from "@vuepress/core";
import { path } from "@vuepress/utils";
import { describe, expect, it } from "vitest";
import { getPageExcerpt } from "vuepress-shared/node";

import { emptyTheme } from "./__fixtures__/theme/empty.js";
import { generatePageIndex } from "../src/node/generateIndex";

const app = createBaseApp({
  bundler: {} as any,
  source: path.resolve(__dirname, "./__fixtures__/src"),
  theme: emptyTheme,
});

await app.init();

describe("generateIndex", () => {
  it("Should generate index", () => {
    app.pages.forEach((page) => {
      page.data["excerpt"] = getPageExcerpt(app, page, {
        excerptLength: 0,
      });

      expect(generatePageIndex(page)).toMatchSnapshot();
    });
  });

  it("Should generate full index", () => {
    app.pages.forEach((page) => {
      page.data["excerpt"] = getPageExcerpt(app, page, {
        excerptLength: 0,
      });

      expect(generatePageIndex(page, [], true)).toMatchSnapshot();
    });
  });

  it("Should support customFields", () => {
    app.pages.forEach((page) => {
      page.data["excerpt"] = getPageExcerpt(app, page, {
        excerptLength: 0,
      });

      expect(
        generatePageIndex(page, [
          {
            getter: ({ frontmatter }: Page): string[] | string | null =>
              (frontmatter.tag as string[] | string) || null,
          },
        ])
      ).toMatchSnapshot();
    });
  });

  it("Should support customFields with full index", () => {
    app.pages.forEach((page) => {
      expect(
        generatePageIndex(
          page,
          [
            {
              getter: ({ frontmatter }: Page): string[] | string | null =>
                (frontmatter.tag as string[] | string) || null,
            },
          ],
          true
        )
      ).toMatchSnapshot();
    });
  });
});
