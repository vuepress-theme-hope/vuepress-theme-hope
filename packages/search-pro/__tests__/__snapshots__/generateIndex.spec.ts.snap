// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateIndex > Should generate full index 1`] = `
{
  "contents": [
    {
      "contents": [
        "Here is article excerpt.",
      ],
      "header": "",
      "slug": "",
    },
    {
      "contents": [
        "Here is main content of article.",
        "A",
        "B",
        "C",
      ],
      "header": "Content",
      "slug": "content",
    },
  ],
  "title": "Demo Page",
}
`;

exports[`generateIndex > Should generate full index 2`] = `
{
  "contents": [
    {
      "contents": [],
      "header": "Heading 2",
      "slug": "heading-2",
    },
    {
      "contents": [],
      "header": "Heading 3",
      "slug": "heading-3",
    },
    {
      "contents": [],
      "header": "Heading 4",
      "slug": "heading-4",
    },
    {
      "contents": [],
      "header": "Heading 5",
      "slug": "heading-5",
    },
    {
      "contents": [],
      "header": "Heading 6",
      "slug": "heading-6",
    },
    {
      "contents": [
        "This sentence has bold、italic and delete style text.",
      ],
      "header": "Text",
      "slug": "text",
    },
    {
      "contents": [
        "This is a paragraph.",
        "This is another paragraph.",
      ],
      "header": "Paragraph",
      "slug": "paragraph",
    },
    {
      "contents": [
        "I would like to line break at this point",
        "::: tip",
        "In codes above, two spaces are behind at.",
        ":::",
      ],
      "header": "Line Break",
      "slug": "line-break",
    },
    {
      "contents": [
        "Blockquotes can also be nested...",
        "...by using greater-than signs right next to each other...",
        "...or with spaces between arrows.",
      ],
      "header": "Blockquotes",
      "slug": "blockquotes",
    },
    {
      "contents": [],
      "header": "List",
      "slug": "list",
    },
    {
      "contents": [
        "Create a list by starting a line with -",
        "Make sub-lists by indenting 2 spaces:",
        "Marker character change forces new list start:",
        "Ac tristique libero volutpat at",
        "Facilisis in pretium nisl aliquet",
        "Nulla volutpat aliquam velit link break",
        "New paragraph",
        "It’s easy!",
      ],
      "header": "Unordered List",
      "slug": "unordered-list",
    },
    {
      "contents": [
        "Lorem ipsum dolor sit amet",
        "Consectetur adipiscing elit line break line break again",
        "Integer molestie lorem at massa",
      ],
      "header": "Ordered List",
      "slug": "ordered-list",
    },
    {
      "contents": [],
      "header": "HR",
      "slug": "hr",
    },
    {
      "contents": [
        "Home page using absolute path",
        "Home page using relative path",
      ],
      "header": "Link",
      "slug": "link",
    },
    {
      "contents": [],
      "header": "Image",
      "slug": "image",
    },
    {
      "contents": [
        "Classic:",
        "😉 😢 😆 😋",
      ],
      "header": "Emoji",
      "slug": "emoji",
    },
    {
      "contents": [
        "center",
        "right",
        "left",
        "For center align use :-:",
        "For right align use -:",
        "For left align use :-",
        "b",
        "aaaaaaaaa",
        "aaaa",
        "c",
        "aaaa",
        "a",
      ],
      "header": "Tables",
      "slug": "tables",
    },
    {
      "contents": [
        "Inline Code: code",
        "Block code:",
        "Syntax highlighting:",
      ],
      "header": "Codes",
      "slug": "codes",
    },
  ],
  "title": "Heading 1",
}
`;

exports[`generateIndex > Should generate full index 3`] = `
{
  "contents": [
    {
      "contents": [
        "Markdown content.",
      ],
      "header": "",
      "slug": "",
    },
  ],
  "title": "Tag Test",
}
`;

exports[`generateIndex > Should generate full index 4`] = `
{
  "contents": [
    {
      "contents": [
        "Markdown content.",
      ],
      "header": "",
      "slug": "",
    },
  ],
  "title": "",
}
`;

exports[`generateIndex > Should generate full index 5`] = `null`;

exports[`generateIndex > Should generate index 1`] = `
{
  "contents": [
    {
      "contents": [
        "Here is article excerpt.",
      ],
      "header": "",
      "slug": "",
    },
  ],
  "title": "Demo Page",
}
`;

exports[`generateIndex > Should generate index 2`] = `
{
  "contents": [
    {
      "contents": [],
      "header": "Heading 2",
      "slug": "heading-2",
    },
    {
      "contents": [],
      "header": "Heading 3",
      "slug": "heading-3",
    },
    {
      "contents": [],
      "header": "Heading 4",
      "slug": "heading-4",
    },
    {
      "contents": [],
      "header": "Heading 5",
      "slug": "heading-5",
    },
    {
      "contents": [],
      "header": "Heading 6",
      "slug": "heading-6",
    },
    {
      "contents": [],
      "header": "Text",
      "slug": "text",
    },
    {
      "contents": [],
      "header": "Paragraph",
      "slug": "paragraph",
    },
    {
      "contents": [],
      "header": "Line Break",
      "slug": "line-break",
    },
    {
      "contents": [],
      "header": "Blockquotes",
      "slug": "blockquotes",
    },
    {
      "contents": [],
      "header": "List",
      "slug": "list",
    },
    {
      "contents": [],
      "header": "Unordered List",
      "slug": "unordered-list",
    },
    {
      "contents": [],
      "header": "Ordered List",
      "slug": "ordered-list",
    },
    {
      "contents": [],
      "header": "HR",
      "slug": "hr",
    },
    {
      "contents": [],
      "header": "Link",
      "slug": "link",
    },
    {
      "contents": [],
      "header": "Image",
      "slug": "image",
    },
    {
      "contents": [],
      "header": "Emoji",
      "slug": "emoji",
    },
    {
      "contents": [],
      "header": "Tables",
      "slug": "tables",
    },
  ],
  "title": "Heading 1",
}
`;

exports[`generateIndex > Should generate index 3`] = `
{
  "contents": [],
  "title": "Tag Test",
}
`;

exports[`generateIndex > Should generate index 4`] = `
{
  "contents": [],
  "title": "",
}
`;

exports[`generateIndex > Should generate index 5`] = `null`;

exports[`generateIndex > Should support customFields 1`] = `
{
  "contents": [
    {
      "contents": [
        "Here is article excerpt.",
      ],
      "header": "",
      "slug": "",
    },
  ],
  "customFields": {
    "0": [
      "Demo",
    ],
  },
  "title": "Demo Page",
}
`;

exports[`generateIndex > Should support customFields 2`] = `
{
  "contents": [
    {
      "contents": [],
      "header": "Heading 2",
      "slug": "heading-2",
    },
    {
      "contents": [],
      "header": "Heading 3",
      "slug": "heading-3",
    },
    {
      "contents": [],
      "header": "Heading 4",
      "slug": "heading-4",
    },
    {
      "contents": [],
      "header": "Heading 5",
      "slug": "heading-5",
    },
    {
      "contents": [],
      "header": "Heading 6",
      "slug": "heading-6",
    },
    {
      "contents": [],
      "header": "Text",
      "slug": "text",
    },
    {
      "contents": [],
      "header": "Paragraph",
      "slug": "paragraph",
    },
    {
      "contents": [],
      "header": "Line Break",
      "slug": "line-break",
    },
    {
      "contents": [],
      "header": "Blockquotes",
      "slug": "blockquotes",
    },
    {
      "contents": [],
      "header": "List",
      "slug": "list",
    },
    {
      "contents": [],
      "header": "Unordered List",
      "slug": "unordered-list",
    },
    {
      "contents": [],
      "header": "Ordered List",
      "slug": "ordered-list",
    },
    {
      "contents": [],
      "header": "HR",
      "slug": "hr",
    },
    {
      "contents": [],
      "header": "Link",
      "slug": "link",
    },
    {
      "contents": [],
      "header": "Image",
      "slug": "image",
    },
    {
      "contents": [],
      "header": "Emoji",
      "slug": "emoji",
    },
    {
      "contents": [],
      "header": "Tables",
      "slug": "tables",
    },
  ],
  "title": "Heading 1",
}
`;

exports[`generateIndex > Should support customFields 3`] = `
{
  "contents": [],
  "customFields": {
    "0": [
      "markdown",
    ],
  },
  "title": "Tag Test",
}
`;

exports[`generateIndex > Should support customFields 4`] = `
{
  "contents": [],
  "customFields": {
    "0": [
      "markdown",
    ],
  },
  "title": "",
}
`;

exports[`generateIndex > Should support customFields 5`] = `null`;

exports[`generateIndex > Should support customFields with full index 1`] = `
{
  "contents": [
    {
      "contents": [
        "Here is article excerpt.",
      ],
      "header": "",
      "slug": "",
    },
    {
      "contents": [
        "Here is main content of article.",
        "A",
        "B",
        "C",
      ],
      "header": "Content",
      "slug": "content",
    },
  ],
  "customFields": {
    "0": [
      "Demo",
    ],
  },
  "title": "Demo Page",
}
`;

exports[`generateIndex > Should support customFields with full index 2`] = `
{
  "contents": [
    {
      "contents": [],
      "header": "Heading 2",
      "slug": "heading-2",
    },
    {
      "contents": [],
      "header": "Heading 3",
      "slug": "heading-3",
    },
    {
      "contents": [],
      "header": "Heading 4",
      "slug": "heading-4",
    },
    {
      "contents": [],
      "header": "Heading 5",
      "slug": "heading-5",
    },
    {
      "contents": [],
      "header": "Heading 6",
      "slug": "heading-6",
    },
    {
      "contents": [
        "This sentence has bold、italic and delete style text.",
      ],
      "header": "Text",
      "slug": "text",
    },
    {
      "contents": [
        "This is a paragraph.",
        "This is another paragraph.",
      ],
      "header": "Paragraph",
      "slug": "paragraph",
    },
    {
      "contents": [
        "I would like to line break at this point",
        "::: tip",
        "In codes above, two spaces are behind at.",
        ":::",
      ],
      "header": "Line Break",
      "slug": "line-break",
    },
    {
      "contents": [
        "Blockquotes can also be nested...",
        "...by using greater-than signs right next to each other...",
        "...or with spaces between arrows.",
      ],
      "header": "Blockquotes",
      "slug": "blockquotes",
    },
    {
      "contents": [],
      "header": "List",
      "slug": "list",
    },
    {
      "contents": [
        "Create a list by starting a line with -",
        "Make sub-lists by indenting 2 spaces:",
        "Marker character change forces new list start:",
        "Ac tristique libero volutpat at",
        "Facilisis in pretium nisl aliquet",
        "Nulla volutpat aliquam velit link break",
        "New paragraph",
        "It’s easy!",
      ],
      "header": "Unordered List",
      "slug": "unordered-list",
    },
    {
      "contents": [
        "Lorem ipsum dolor sit amet",
        "Consectetur adipiscing elit line break line break again",
        "Integer molestie lorem at massa",
      ],
      "header": "Ordered List",
      "slug": "ordered-list",
    },
    {
      "contents": [],
      "header": "HR",
      "slug": "hr",
    },
    {
      "contents": [
        "Home page using absolute path",
        "Home page using relative path",
      ],
      "header": "Link",
      "slug": "link",
    },
    {
      "contents": [],
      "header": "Image",
      "slug": "image",
    },
    {
      "contents": [
        "Classic:",
        "😉 😢 😆 😋",
      ],
      "header": "Emoji",
      "slug": "emoji",
    },
    {
      "contents": [
        "center",
        "right",
        "left",
        "For center align use :-:",
        "For right align use -:",
        "For left align use :-",
        "b",
        "aaaaaaaaa",
        "aaaa",
        "c",
        "aaaa",
        "a",
      ],
      "header": "Tables",
      "slug": "tables",
    },
    {
      "contents": [
        "Inline Code: code",
        "Block code:",
        "Syntax highlighting:",
      ],
      "header": "Codes",
      "slug": "codes",
    },
  ],
  "title": "Heading 1",
}
`;

exports[`generateIndex > Should support customFields with full index 3`] = `
{
  "contents": [
    {
      "contents": [
        "Markdown content.",
      ],
      "header": "",
      "slug": "",
    },
  ],
  "customFields": {
    "0": [
      "markdown",
    ],
  },
  "title": "Tag Test",
}
`;

exports[`generateIndex > Should support customFields with full index 4`] = `
{
  "contents": [
    {
      "contents": [
        "Markdown content.",
      ],
      "header": "",
      "slug": "",
    },
  ],
  "customFields": {
    "0": [
      "markdown",
    ],
  },
  "title": "",
}
`;

exports[`generateIndex > Should support customFields with full index 5`] = `null`;
